### PR TITLE
Fix model name with whitespace on keyboard content-builder form submission

### LINKED ISSUE #1897: Fail to create new content when its value of 'Name' field contains white-spaces
**Informations**
- **Node.js version**: 10.9.0
- **npm version**: 6.2.0
- **Strapi version**: 3.0.0-alpha.14
- **Database**: MongoDB v3.4.16
- **Operating system**: CentOS Linux release 7.5.1804 (on AWS Lightsail) 

**What is the current behavior?**
Once I created content type w/ 'Name' field contains any white-spaces at Content Type Builder, then it freeze completely at loading status when I try to add new content. 

**What is the expected behavior?**
Preferred: Warn 'Name contains invalid character' or something before create the content type at Builder, and fail to create record.

**Suggested solutions**
Insert no spaces when input a name of content type.

### PR DESCRIPTION
**Pull request by Tonours**
My PR is a:
🐛 Bug fix #1897 

Main update on the:
Update of the `strapi-plugin-content-type-builder`

Model name is camelCased only on input blur, a `getModelWithCamelCaseName` function has been added and is used for model object on form submission (create and edit) in order to prevent model name with whitespaces.

### CODE REVIEW

File: packages/strapi-plugin-content-type-builder/admin/src/containers/Form/index.js [hunk 1]
```diff
@@ -514,9 +526,12 @@ export class Form extends React.Component { // eslint-disable-line react/prefer-
...
 
   renderModalBodyChooseAttributes = () => {
-    const attributesDisplay = has(this.context.plugins.toJS(), 'upload') 
-      ? forms.attributesDisplay.items
-      : forms.attributesDisplay.items.filter(obj => obj.type !== 'media'); // Don't display the media field if the upload plugin isn't installed
+    const attributesDisplay = forms.attributesDisplay.items;
+
+    // Don't display the media field if the upload plugin isn't installed
+    if (!has(this.context.plugins.toJS(), 'upload')) {
+      attributesDisplay.splice(8, 1);
```
Comment (soupette) [inline packages/strapi-plugin-content-type-builder/admin/src/containers/Form/index.js]: You might need to clone the `attributesDisplay ` before splicing it. I remember that I used to do the same logic and it caused the attributesForm to be buggy.

```
const attributesDisplay = (() => {
  const list = forms.attributesDisplay.items;

  if (!this.context.plugins.toJS().hasOwnProperty('upload')) {
     return list.filter(obj => obj.type !== 'media';
  }
   return list;
})();
```

File: packages/strapi-plugin-content-type-builder/admin/src/containers/Form/index.js [hunk 2]
```diff
@@ -446,17 +446,16 @@ export class Form extends React.Component { // eslint-disable-line react/prefer-
...
           break;
       }
-  
+
       this.setState(prevState => ({ nodeToFocus: prevState.nodeToFocus + toAdd }));
     }
   }
 
-  handleSubmit = (e, redirectToChoose = true) => {
+  handleSubmit = (e, redirectToChoose = false) => {
```
Comment (soupette) [inline packages/strapi-plugin-content-type-builder/admin/src/containers/Form/index.js]: I'm wondering why you changed the redirectToChoose param?

Comment (Tonours) [reply packages/strapi-plugin-content-type-builder/admin/src/containers/Form/index.js]: Yes it shouldn't i had issue with syncing my fork, put code back.

File: packages/strapi-plugin-content-type-builder/admin/src/containers/Form/index.js (hunk 1 above)
Comment (Tonours) [reply packages/strapi-plugin-content-type-builder/admin/src/containers/Form/index.js]: Same for here,  it shouldn't be update, i put code back here too

**Pull request merged**